model/host/host_intent.go

```diff
@@ -78,6 +78,37 @@ func NewIntent(d distro.Distro, instanceName, provider string, options CreateOpt
 	return intentHost
 }
 
+func newIntentFromHost(h *Host) *Host {
+	newID := h.Distro.GenerateName()
+	intentHost := &Host{
```

Comment:
This struct initialization seems really similar to `NewIntent` in that both set the same fields in much the same way to achieve the same goal. Can we make both of them call the same common method to create the new host model to keep them in agreement on how to create an intent host?

Reply:
I think this is good now.